src/Database/TypeMap.php

```diff
@@ -17,144 +17,61 @@
-     *
-     * Used to avoid repetition when calling multiple functions inside this class that
-     * may require a custom type for a specific field.
-     *
-     * @var array<string, string>
-     */
-    protected array $_types = [];
-
-    /**
-     * Creates an instance with the given defaults
-     *
-     * @param array<string, string> $defaults The defaults to use.
-     */
-    public function __construct(array $defaults = [])
-    {
-        $this->setDefaults($defaults);
-    }
-
-    /**
-     * Configures a map of fields and associated type.
-     *
-     * These values will be used as the default mapping of types for every function
-     * in this instance that supports a `$types` param.
-     *
-     * This method is useful when you want to avoid repeating type definitions
-     * as setting types overwrites the last set of types.
-     *
-     * ### Example
-     *
-     * ```
-     * $query->setDefaults(['created' => 'datetime', 'is_visible' => 'boolean']);
-     * ```
-     *
-     * This method will replace all the existing default mappings with the ones provided.
-     * To add into the mappings use `addDefaults()`.
-     *
-     * @param array<string, string> $defaults Associative array where keys are field names and values
-     * are the correspondent type.
-     * @return $this
+     * @var array<string>
      */
-    public function setDefaults(array $defaults)
-    {
-        $this->_defaults = $defaults;
-
-        return $this;
-    }
+    protected array $_types;
 
     /**
-     * Returns the currently configured types.
-     *
-     * @return array<string, string>
+     * @param array<string> $types Map of fields to types
      */
-    public function getDefaults(): array
+    public function __construct(array $types = [])
     {
-        return $this->_defaults;
-    }
-
-    /**
-     * Add additional default types into the type map.
-     *
-     * If a key already exists it will not be overwritten.
-     *
-     * @param array<string, string> $types The additional types to add.
-     * @return void
-     */
-    public function addDefaults(array $types): void
-    {
-        $this->_defaults += $types;
+        $this->_types = $types;
     }
 
     /**
-     * Sets a map of fields and their associated types for single-use.
+     * Sets types for fields.
      *
-     * ### Example
+     * If a field is already mapped, the type is replaced.
      *
+     * Example:
      * ```
-     * $query->setTypes(['created' => 'time']);
+     * $typeMap->setTypes(['id' => 'integer']);
      * ```
      *
-     * This method will replace all the existing type maps with the ones provided.
-     *
-     * @param array<string, string> $types Associative array where keys are field names and values
-     * are the correspondent type.
+     * @param array<string> $types Map of fields to types
      * @return $this
      */
     public function setTypes(array $types)
```

Comment:
Should we leave this as an 'overwrite' to preserve backwards compat more? We could have an `addTypes()` method to do the merge add instead.

Reply:
I was either going to do that or add an 'overwrite' parameter.

Reply:
More methods is better. Merge/overwrite parameters have not aged well.

Reply:
I don't know if it happens anywhere, but merging int columns doesn't work. They can basically only be appended. I think that's how it's always expected to happen, but our API kind of hides that.